### PR TITLE
JA.1: Employee CRUD and Basic Scaffolding

### PR DESCRIPTION
**Pull request by kentherussell**
Initial Scaffolding. Some exceptions and situations aren't handled but I will add more tickets for those on Trello.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -26,19 +26,27 @@ repositories {
 
 dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter-actuator'
-	implementation 'org.springframework.boot:spring-boot-starter-jdbc'
+//	implementation 'org.springframework.boot:spring-boot-starter-jdbc'
```
[the remark this answers is not in the document]
Comment (kentherussell) [reply build.gradle]: Thanks for the heads up, will change.

File: build.gradle [hunk 2]
```diff
@@ -26,19 +26,27 @@ repositories {
...
+
 	implementation 'org.springframework.boot:spring-boot-starter-jetty'
 	implementation('org.springframework.boot:spring-boot-starter-web') {
 		exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
 	}
+
 	implementation 'org.springframework.boot:spring-boot-starter-webflux'
 	implementation 'org.springframework.boot:spring-boot-starter-logging'
-	implementation 'org.springframework.data:spring-data-jdbc'
+//	implementation 'org.springframework.data:spring-data-jdbc'
```
[the remark this answers is not in the document]
Comment (kentherussell) [reply build.gradle]: Removed

File: build.gradle [hunk 3]
```diff
@@ -26,19 +26,27 @@ repositories {
...
 	implementation 'org.flywaydb:flyway-core'
 	implementation 'io.springfox:springfox-swagger2:2.9.2'
 	implementation 'io.springfox:springfox-swagger-ui:2.9.2'
 	implementation 'net.logstash.logback:logstash-logback-encoder:5.3'
 	implementation 'org.springframework.boot:spring-boot-configuration-processor'
+	implementation 'javax.persistence:javax.persistence-api:2.2'
+	implementation 'org.springframework.boot:spring-boot-devtools'
```
[the remark this answers is not in the document]
Comment (kentherussell) [reply build.gradle]: Thanks for the heads up, just changed this.

File: build.gradle [hunk 4]
```diff
@@ -18,27 +18,38 @@ configurations {
...
 
+
 repositories {
 	mavenCentral()
 }
 
 dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter-actuator'
-	implementation 'org.springframework.boot:spring-boot-starter-jdbc'
+	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
```
Comment (vickumar1981) [inline build.gradle]: spring-data-jdbc is certainly the newer repo.  For postgres, this change doesn't make too much difference now, but for Mongo and Cassandra, data-jdbc is reactive where jpa is not.  also, i believe there are plans on making postgres truly reactive, but that won't be in the jpa version.

File: build.gradle [hunk 5]
```diff
@@ -18,27 +18,38 @@ configurations {
...
 	implementation 'org.springframework.boot:spring-boot-starter-logging'
-	implementation 'org.springframework.data:spring-data-jdbc'
 	implementation 'org.flywaydb:flyway-core'
 	implementation 'io.springfox:springfox-swagger2:2.9.2'
 	implementation 'io.springfox:springfox-swagger-ui:2.9.2'
 	implementation 'net.logstash.logback:logstash-logback-encoder:5.3'
 	implementation 'org.springframework.boot:spring-boot-configuration-processor'
+	implementation 'javax.persistence:javax.persistence-api:2.2'
+
+	developmentOnly "org.springframework.boot:spring-boot-devtools"
```
Comment (vickumar1981) [inline build.gradle]: is this for hot-swapping?  i suspect there's more to do, to get hot-swapping working inside the docker container.

File: src/main/java/com/excella/reactor/domain/Employee.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@NoArgsConstructor
+@Entity
+public class Employee extends DomainModel {
+
+  @Embedded private Bio bio;
+
+  @Embedded private Contact contact;
```
Comment (vickumar1981) [inline src/main/java/com/excella/reactor/domain/Employee.java]: what about the address?

File: start [hunk 7]
```diff
@@ -1,2 +1,3 @@
+docker-compose down
```
Comment (vickumar1981) [inline start]: i feel like this could just be in a separate script - maybe `stop` or something.

**Pull request merged**